### PR TITLE
feat(site): add ability to create tokens from account tokens page

### LINKED ISSUE #5778: tokens: support creation from UI
Add all fields from CLI (expiration date, token name, etc). GitHub or Gitlab can be referenced for the UI for picking expiration, they both have slightly different styles.

Screen Shot 2023-01-18 at 2 43 45 PM

### PR DESCRIPTION
**Pull request by Kira-Pilot**
resolves #5778

Some notes:
- We are stuck using a native datepicker (wrapped in the MUI `TextField` component) as we are on an older version of MUI; as such, styling options are limited.
- The [original mocks](https://www.figma.com/file/oXAFu6Z4Pmv7GGux16ZjAK/Token-Creation?node-id=0%3A1&t=RQuYKUHD20qxeACd-0) for this design had a "no expiration" option. This is something I'd like to add down the road (here's a [ticket](https://github.com/coder/coder/issues/6607)) but right now we do not support this option in the API or the CLI, and there's some plumbing involved that warrants its own PR.

**Token creation without `--max-token-lifetime set**
Kapture 2023-03-14 at 16 59 53

**Token creation with a `--max-token-lifetime` of 20 days set**
Kapture 2023-03-14 at 17 02 52

### CODE REVIEW

File: site/src/pages/CreateTokenPage/utils.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    value: 90,
+  },
+]
+
+export const customLifetimeDay: LifetimeDay = {
+  label: i18next.t("tokensPage:createToken.lifetimeSection.custom"),
+  value: "custom",
+}
+
+export const filterByMaxTokenLifetime = (
```
Comment (Kira-Pilot) [inline site/src/pages/CreateTokenPage/utils.ts:39]: Will write some unit tests for these helper functions

File: site/src/pages/CreateTokenPage/CreateTokenPage.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import { CreateTokenForm } from "./CreateTokenForm"
+import { NANO_HOUR, CreateTokenData } from "./utils"
+import { AlertBanner } from "components/AlertBanner/AlertBanner"
+
+const initialValues: CreateTokenData = {
+  name: "",
+  lifetime: 30,
+}
+
+const CreateTokenPage: FC = () => {
```
Comment (Kira-Pilot) [inline site/src/pages/CreateTokenPage/CreateTokenPage.tsx:21]: Will add some visual stories/tests for this new form

File: site/src/pages/CreateTokenPage/utils.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import i18next from "i18next"
+
+export const NANO_HOUR = 3600000000000
```
Comment (Kira-Pilot) [inline site/src/pages/CreateTokenPage/utils.ts:3]: This is a bit of a pain - explanation [here](https://github.com/coder/coder/issues/6607).

**Review by BrunoQuaresma — APPROVED** [review]
Awesome work!

**Pull request merged**